File: http/src/main/java/com/baihe/http/JsonParam.java
package com.baihe.http;

import android.text.TextUtils;

import com.alibaba.fastjson.JSONObject;

import java.util.Map;

/**
 * Author：xubo
 * Time：2018-12-28
 * Description：json参数
 */
public class  JsonParam {
    private String paramKey;
    private JSONObject paramValueJSONObject;

    public static JsonParam newInstance(String paramKey) {
        return new JsonParam(paramKey);
    }

    public JsonParam(String paramKey) {
        if (TextUtils.isEmpty(paramKey)) {
            throw new IllegalStateException("json参数的key不能为空");
        }
        this.paramKey = paramKey;
        this.paramValueJSONObject = new JSONObject();
    }

    /**
     * json参数的添加具体值
     *
     * @param key
     * @param value
     * @return
     */
    public JsonParam putParamValue(String key, Object value) {
        if (!TextUtils.isEmpty(key) && value != null) {
            paramValueJSONObject.put(key, value);
        }
        return this;
    }

    /**
     * json参数的添加具体值
     *
     * @param paramvalues
     * @return
     */
    public JsonParam putParamValue(Map<String, Object> paramvalues) {
        if (paramvalues != null) {
            paramValueJSONObject.putAll(paramvalues);
        }
        return this;
    }

    /**
     * 获取json参数的key
     *
     * @return
     */
    public String getParamKey() {
        return paramKey;
    }

    /**
     * 获取json参数的value
     *
     * @return
     */
    public String getParamValue() {
        return paramValueJSONObject.toJSONString();
    }
}

File: Lib_Common/src/main/java/com/baihe/lib_common/http/api/JsonParam.kt
package com.baihe.lib_common.http.api

import android.text.TextUtils
import com.google.gson.GsonBuilder

class JsonParam()  {
    private val params: MutableMap<String, Any?> = mutableMapOf()
    private val gson by lazy {
        GsonBuilder().setLenient()// json宽松
            .enableComplexMapKeySerialization()//支持Map的key为复杂对象的形式
            .serializeNulls() //智能null
            .setPrettyPrinting()// 调教格式
            .disableHtmlEscaping() //默认是GSON把HTML 转义的
            .create()
    }


    companion object {
        fun newInstance():JsonParam{
            return JsonParam()
        }
    }


    fun putParamValue(key: String?, value: Any?): JsonParam {
        if(!key.isNullOrEmpty() && value!=null){
            params[key]=value
        }
        return this
    }

    fun putParamValue(paramValues: LinkedHashMap<String, Any?>?): JsonParam {
        if (!paramValues.isNullOrEmpty()) {
            params.putAll(paramValues)
        }
        return this
    }
    fun getParamValue(): String? {
        return gson.toJson(params)
    }
}